app-backend/db/src/main/scala/com/azavea/rf/database/DatasourceDao.scala

```diff
@@ -28,28 +31,70 @@ object DatasourceDao extends Dao[Datasource] {
     """ ++ tableF
 
   def create(
-    user: User,
-    organizationId: UUID,
-    name: String,
-    visibility: Visibility,
-    owner: Option[String],
-    composites: Json,
-    extras: Json,
-    bands: Json
+    datasource: Datasource,
+    user: User
   ): ConnectionIO[Datasource] = {
-    val id = UUID.randomUUID
-    val now = new Timestamp((new java.util.Date()).getTime())
-    val ownerId = util.Ownership.checkOwner(user, owner)
+    val ownerId = util.Ownership.checkOwner(user, Some(datasource.owner))
     (fr"INSERT INTO" ++ tableF ++ fr"""
       (id, created_at, created_by, modified_at, modified_by, owner,
       organization_id, name, visibility, composites, extras, bands)
     VALUES
-      ($id, $now, ${user.id}, $now, ${user.id}, $ownerId,
-      $organizationId, $name, $visibility, $composites, $extras, $bands)
+      (${datasource.id}, ${datasource.createdAt}, ${datasource.createdBy}, ${datasource.modifiedAt},
+      ${datasource.modifiedBy}, ${ownerId}, ${datasource.organizationId}, ${datasource.name},
+      ${datasource.visibility}, ${datasource.composites},
+      ${datasource.extras}, ${datasource.bands})
     """).update.withUniqueGeneratedKeys[Datasource](
       "id", "created_at", "created_by", "modified_at", "modified_by", "owner",
       "organization_id", "name", "visibility", "composites", "extras", "bands"
     )
   }
+
+  def updateDatasource(datasource: Datasource, id: UUID, user: User)
+                      (implicit xa: Transactor[IO]): Future[Int] = {
+    // fetch datasource so we can check if user is allowed to update (access control)
+    val now = new Timestamp((new java.util.Date()).getTime())
+    val updateQuery =
+      fr"UPDATE" ++ this.tableF ++ fr"SET" ++
+      fr"""
+      modified_at = ${now},
+      modified_by = ${user.id},
+      name = ${datasource.name},
+      visibility = ${datasource.visibility},
+      composites = ${datasource.composites},
+      extras = ${datasource.extras},
+      bands = ${datasource.bands}
+      where id = $id
```

Comment:
I think we should try and be consistent with formatting -- I think it makes sense to _alway_ use `${}` since it is more explicit and a little less brittle when refactoring.

What do you think?

Reply:
makes sense. Emacs will highlight `$id` and `$x.id` differently, but not all editors will